pennylane/ops/op_math/composite.py

```diff
@@ -219,7 +218,9 @@ def overlapping_ops(self) -> list[list[Operator]]:
                     groups[first_group_idx][1] = groups[first_group_idx][1] + wires
                 else:
                     i += 1
-            if first_group_idx is None:
+            if first_group_idx is not None:
```

Comment:
The `elif` block above merges groups that both overlap with the new op, but it happens after the new op has already been added to the first group, which made it so that the new op is now placed before the ops that originally precedes it. With this fix, the new op is added to the target group after the existing groups are merged into the first group, maintaining the original order of operations.